### PR TITLE
add pyramid_deform_conv to mmcv.op

### PR DESCRIPTION
**Pull request by crazysboy**
我在ops中添加一种pyramid_deform_conv，为此添加和修改了相关必要的程序，如pyramid_deform_conv_cuda_kernel.cuh、pyramid_deform_conv_cuda.cu、cudabind.cpp、pybind.cpp、pyramid_deform_conv.cpp和pyramid_deform_conv.py。虽然能编译成功但测试时会出现类似ImportError: /media/zhaozhenxi/新加卷/papper5/mmcv-master(1)/mmcv-master/mmcv/_ext.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Z19deform_conv_forwardN2at6TensorES0_S0_S0_S0_S0_iiiiiiiiiii。请你版本给我看看那里出现问题，我添加修改的程序是否正确。我添加修改的程序是参照原来中的deform_conv来写，这样能至少能编译成功，按照readme 来会直接失败。我不清楚这些错误该怎么修改。请你帮我检查一下

Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by crazysboy** [comment]
你好，请问这是在检查吗。第一次弄，还不太懂这个PR。

File: mmcv/ops/csrc/pytorch/cuda/cudabind.cpp [hunk 1]
```diff
@@ -319,6 +319,66 @@ REGISTER_DEVICE_IMPL(deformable_col2im_impl, CUDA, deformable_col2im_cuda);
 REGISTER_DEVICE_IMPL(deformable_col2im_coord_impl, CUDA,
                      deformable_col2im_coord_cuda);
 
+
+
+void pyramid_deformable_im2col_cuda(Tensor data_im, Tensor data_offset,
```
Comment (zhouzaida) [inline mmcv/ops/csrc/pytorch/cuda/cudabind.cpp:324]: Hi, I found the parameters here is not consistent with `mmcv/ops/csrc/pytorch/cuda/pyramid_deform_conv_cuda.cu`

https://github.com/open-mmlab/mmcv/pull/1647/files#diff-df24591022d01d0437b10957bdbf2d6364d4618bfbddf27e3d3d0b813807ed26R5-R13

**Comment by crazysboy** [comment]
感谢指出。问题已解决

**Comment by zhouzaida** [comment]
> 感谢指出。问题已解决

可以完善一下 PR，后面我们可以合入 master 分支

**Comment by zhouzaida** [comment]
Is there any progress?

**Comment by crazysboy** [comment]
还不打算先合并。



---Original---
From: "Zaida ***@***.***&gt;
Date: Sat, Jan 29, 2022 00:02 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [open-mmlab/mmcv] add pyramid_deform_conv to mmcv.op (PR #1647)




 
Is there any progress?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by FDInSky** [comment]
> 感谢指出。问题已解决

请问怎么解决的？

File: mmcv/ops/csrc/pytorch/cuda/cudabind.cpp (hunk 1 above)
Comment (FDInSky) [reply mmcv/ops/csrc/pytorch/cuda/cudabind.cpp:324]: 请问是哪里不一致？

**Comment by OpenMMLab-Assistant003** [comment]
Hi @crazysboy！We are grateful for your efforts in helping improve this open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤ @crazysboy

**Pull request closed without merging**